app/controllers/concerns/willow_sword/process_deposit.rb

```diff
@@ -69,6 +69,10 @@ def process_data
         # process xml
         # puts 'process xml'
         process_xml
+      when 'text/xml'
```

Comment:
we could modify the previous when statement to 
```when 'application/xml', 'text/xml'```
and remove this blob

Reply:
Good idea! I've made that change and rebased it into the same commit.